clicommand/agent_start.go

```diff
@@ -798,15 +799,19 @@ func handlePoolSignals(l logger.Logger, pool *agent.AgentPool) chan os.Signal {
 // if it's available. This is very similar to the bootstrap hook code. Just run at
 // the agent level.
 func shutdownHook(log logger.Logger, hooksPath string) {
-	p := filepath.Join(hooksPath, "shutdown")
-	if _, err := os.Stat(p); os.IsNotExist(err) {
+	p, err := hook.Find(hooksPath, "shutdown")
```

Comment:
Here's the crux of the change — `hook.Find` searches `.BAT;.CMD;.PS1` using [the Windows version of `shell.LookPath()`](https://github.com/buildkite/agent/blob/99d709436d9b2894f5e94727d261600831aa1269/bootstrap/shell/lookpath_windows.go#L59).